src/components/Table/GridTable.tsx

```diff
@@ -284,11 +292,26 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     return rows;
   }, [columns, rows, sorting, sortState]);
 
-  const columnSizes = useMemo(
-    () => calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth),
-    [columns, style.nestedCards?.firstLastColumnWidth],
+  const [columnSizes, setColumnSizes] = useState<string[]>(
+    calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth, undefined, style.minWidth),
   );
 
+  const [tableWidth, setTableWidth] = useState<number | undefined>();
```

Comment:
Maybe put this before `const [columnSizes, setColumnSzies] = ` so that we could pass `tableWidth`, even it's initial `undefined` value, to `calcColumnSizes`? Really minor just thinking it might be nice to always see that 3rd param is `tableWidth`.